cpapp_backend/cpapp/services/agent.py

```
@@ -4479,29 +5120,126 @@ def handle_pan_card_number(self, pan_number: str, session_id: str) -> dict:
+                if date_of_birth:
+                    # Convert from DD-MM-YYYY to YYYY-MM-DD format if needed
+                    try:
+                        if '-' in date_of_birth and len(date_of_birth.split('-')) == 3:
+                            parts = date_of_birth.split('-')
+                            if len(parts[2]) == 4:  # DD-MM-YYYY format
+                                pan_details['dateOfBirth'] = f"{parts[2]}-{parts[1]}-{parts[0]}"
+                            else:  # Already in YYYY-MM-DD or other format
+                                pan_details['dateOfBirth'] = date_of_birth
+                        else:
+                            pan_details['dateOfBirth'] = date_of_birth
+                    except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)